### PR TITLE
Mac Integration

### PR DESCRIPTION
**Pull request by xiamubobby**
image

### CODE REVIEW

**Comment by xiamubobby** [comment]
---
```Kotlin
class Test {
    @Test
    fun testXTPLibraryLoaded() {
        TraderApi.CreateTraderApi(1, "./log", XTP_LOG_LEVEL.XTP_LOG_LEVEL_INFO).apply {
            assertEquals("2.2.27", GetApiVersion())
        }
    }
}
```
---

- [ x ] passed

---

方便的话请尽快publish一个1.1.0之类的上mvncentral吧，这样利用起来会方便一些了，眼下我用的静态reference。

**Comment by RationalityFrontline** [comment]
Could you please offer me a already built "libjxtp.dylib" file? I don't have a mac pc, so I can't build the .dylib file and publish it to MavenCentral. You can offer it by attaching the file as attachment in the reply.

**Comment by xiamubobby** [comment]
[jxtp-2.2.27.4-1.0.0.zip](https://github.com/RationalityFrontline/jxtp/files/6351600/jxtp-2.2.27.4-1.0.0.zip)

File: cpp/build.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 rm -rf build
 cmake CMakeLists.txt -B build
 cmake --build build --config Release
-cp build/lib/libjxtp.so ../lib
+lib_ext=".so"
+lib_ext=$([[ $(uname) = "Darwin" ]] && echo ".dylib")
```
Comment (RationalityFrontline) [inline cpp/build.sh]: The lib_ext will be empty under Linux. Did I miss something? The following approach works fine for me:
```suggestion
if [[ "$(uname -s)" == "Darwin" ]] ; then
  lib_ext=".dylib"
fi
```

**Review by RationalityFrontline — COMMENTED** [review]
Maybe the dynamic extension forging didn't work under Linux

**Comment by xiamubobby** [comment]
> Maybe the dynamic extension forging didn't work under Linux

Apologies. Craved for the one-liner but got it wrong.

```Bash
sh-5.1# uname -s
Linux
sh-5.1# echo $([[ "$(uname -s)" = "Darwin"  ]] && echo ".dylib" || echo ".so")
.so
```

```Bash
bash-3.2$ uname -s
Darwin
bash-3.2$  echo $([[ "$(uname -s)" = "Darwin"  ]] && echo ".dylib" || echo ".so")
.dylib
```

Fixed.

**Pull request merged**